### ISSUE TITLE
I can't find responsive images?

### ISSUE THREAD

**Issue by xaiyeon** [Ks89/angular-modal-gallery#129]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ x] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior

N/A

## Expected behavior

I was reading the documentation and it's nice, but hard to follow if you don't start with the examples and demos. However, is there a way for responsive images for the PlainGallery mode and grids?

## Minimal reproduction of the problem with instructions

My official live example on StackBlitz (based on https://stackblitz.com/edit/angular-modal-gallery-v5) is: 

## What is the motivation / use case for changing the behavior?

So that the pictures can resize and also resize on mobile like Bootstrap responsive images.

## Environment (the most important section to fill very carefully)

<pre><code>
- Node version: X.X.X  
- npm version: X.X.X  
- Operating System and version:  
- Angular version: X.Y.Z 
- angular-cli version (or SystemJS/Webpack): X.Y.Z 
- I'm using Server Side Rendering with angular-universal: YES/NO
- I'm compiling with mode: DEBUG / PROD / PROD with AOT

Browser:
- [x ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [x ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

Others:

</code></pre>

**Comment by Ks89** [comment]
Plain gallery is made to show small images and clicking on one of their you'll see the modal image in full screen. Why would you show a plain image in a responsive way filling the screen horizontally?

However, as in my official demos, for instance on stackblitz, you can draw your custom plain gallery iterating over images with an ngfor using your css.

**Comment by Ks89** [comment]
PS: do you have any suggestions to improve the doc?
Feel free to propose your ideas, don't be shy :)

**Comment by xaiyeon** [comment]
Hi, first, thanks very much for making this library! It's for the mobile part when the images get smaller, that's why, it just bothers me that the photo is small and doesn't fill up space. I looked at stackblitz for example Z4, Z5, and Z6, is it possible to add a style class to it? When you define the Layout:

something like:
`layout: new LineLayout({ width: 'flex', height: 'auto' }, { length: 4, wrap: true }, 'img-responsive'),`

Also in mobile, the slideshow isn't great on mobile, but you probably know that. I'll use an ngif on ngtemplate for that if detect on mobile to switch templates.

I was wondering in the documentation you can add drop-down menu for your Features like PlainGallery has a dropdown and the features inside that instead of clicking and scrolling and looking. Also an example code, that refers to the feature as an example (it takes the time out of scrolling through the stackblitz and reading and looking for what the code does).

**Comment by Ks89** [comment]
```
I was wondering in the documentation you can add drop-down menu for your Features like PlainGallery has a dropdown and the features inside that instead of clicking and scrolling and looking.
```
Yes this is a good idea, I'll add it in my schedule for next releases.


```
Also an example code, that refers to the feature as an example (it takes the time out of scrolling through the stackblitz and reading and looking for what the code does).
```
I don't understand this, are you saying to add a link to the related demo for that particular feature? If yes, it's also a good idea (some examples are cross-features, so It will be dificult, but I can try it)


```
Also in mobile, the slideshow isn't great on mobile, but you probably know that. I'll use an ngif on ngtemplate for that if detect on mobile to switch templates.
```

Do you have an idea of how to improve it or do you want to contribute proposing a new layout? If you don't want to write code, I'll accept also sketches and mocks in paint :)...but I prefer contributions, in particular this Spring, because I'm really busy :(.


```
Hi, first, thanks very much for making this library! It's for the mobile part when the images get smaller, that's why, it just bothers me that the photo is small and doesn't fill up space. I looked at stackblitz for example Z4, Z5, and Z6, is it possible to add a style class to it? When you define the Layout:
something like:
layout: new LineLayout({ width: 'flex', height: 'auto' }, { length: 4, wrap: true }, 'img-responsive'),
```

I really don't understand this. Plain gallery is used to show many images on the screen (you can choose the size), not to show it in full screen like modal gallery.
I don't want to add something like bootstrap images in plain gallery, because the purpose of plain gallery is totally different. Modal gallery shows images in full screen, but plain gallery not.
Z4, Z5 and Z6 are completely custom so you can use you classes as you prefer. When you use custom plain gallery you have to use AdvancedLayout, not LineLayout that is build to be very basic.
As you can see in my code the plain gallery is configured in this way:
```
customPlainGalleryRowDescConfig: PlainGalleryConfig = {
    strategy: PlainGalleryStrategy.CUSTOM,
    layout: new AdvancedLayout(-1, true)
  };
```
but the gallery appears thanks to this code:
```
<div class="my-app-custom-plain-container-column">
    <ng-container *ngFor="let img of images; let i = index">
      <div *ngIf="i <= 2"> <!-- 2 is a fixed value that I'm using to display only two images -->
        <a class="more" *ngIf="i==2" (click)="openImageModalColumn(img)"> +{{images.length - 2 - 1}} more </a>
        <img *ngIf="img.plain && img.plain.img; else noThumb"
             class="my-app-custom-image-column"
             [src]="img.plain.img"
             (click)="openImageModalColumn(img)"
             [alt]="img.modal.description"/>

        <ng-template #noThumb>
          <img class="my-app-custom-image-column"
               [src]="img.modal.img"
               (click)="openImageModalColumn(img)"
               [alt]="img.modal.description"/>
        </ng-template>
      </div>
    </ng-container>
  </div>
```
As you can see you can write the layout and the behaviour that you prefer, because I'm using my classes (like 'my-app-custom-plain-container-column', 'my-app-custom-image-column'...) . Feel free to be creative.

**Comment by Ks89** [comment]
I saved on a private to list some info about this issues, so I can close it right now.

Thanks
